cachito/web/api_v1.py

```diff
@@ -74,6 +77,28 @@ def get_requests():
         state_int = RequestStateMapping.__members__[state].value
         query = query.join(RequestState, Request.request_state_id == RequestState.id)
         query = query.filter(RequestState.state == state_int)
+    repo = flask.request.args.get("repo")
+    if repo:
+        query = query.filter(Request.repo == repo)
+    ref = flask.request.args.get("ref")
+    if ref:
+        if not is_request_ref_valid(ref):
+            raise ValidationError(f"{ref} is not a valid ref.")
+        query = query.filter(Request.ref == ref)
+    pkg_managers = flask.request.args.getlist("pkg_manager")
+    if pkg_managers:
+        qr = (
+            PackageManager.query.filter(PackageManager.name.in_(pkg_managers))
+            .with_entities(PackageManager.id)
+            .all()
+        )
+        pkg_manager_ids = [pm_id for pm_id, in qr]
```

Comment:
Could this be a method of the `PackageManager` class? Something like `pkg_manager_ids = [PackageManager.get_id(pm_name) for pm_name in pkg_managers]`. You could even decorate that method with `functools.cache` to make sure the DB will never be queried more than once for the same package manager.